Source/WTF/wtf/NativePromise.h

```diff
@@ -546,13 +552,14 @@ class NativePromise final : public NativePromiseBase, public ConvertibleToNative
             }
             m_producer->reject(std::forward<RejectValueType_>(rejectValue));
             m_producer = nullptr;
-            m_resolveValues.clear();
+            if constexpr (!std::is_void_v<ResolveValueT>)
+                m_resolveValues.clear();
         }
 
         Ref<AllPromiseType> promise() { return static_cast<Ref<AllPromiseType>>(*m_producer); }
 
     private:
-        Vector<std::optional<ResolveValueType>> m_resolveValues;
+        NO_UNIQUE_ADDRESS std::conditional_t<!std::is_void_v<ResolveValueType>, Vector<std::optional<ResolveValueType>>, std::monostate> m_resolveValues;
```

Comment:
`std::monostate` is declared in `<variant>`, you should include it to avoid potential broken builds.
(I've just noticed it was already used in this file, missed that in previous reviews.)

Optional alternative: It feels a bit like an abuse of that type that is intended for use in `std::variant`.
Since they're not publicly visible here, would it be possible to replace them with private empty types?

Reply:
std::monostate seems to have been used in combination of NO_UNIQUE_ADDRESS to generate a 0-bytes member. I only followed existing examples.

I'm not sure on which to use otherwise.

Reply:
I was suggesting you create your own empty type right above this line. 😉
Something like:
```
struct VoidPlaceholder {};
NO_UNIQUE_ADDRESS std::conditional_t<!std::is_void_v<ResolveValueType>, Vector<std::optional<ResolveValueType>>, VoidPlaceholder> m_resolveValues;
```